test/runtests.jl

```diff
@@ -25,22 +25,34 @@ using Test
     println("Running `tmap!` test...")
     @test tmap!(foo, z, x, y) ≈ foo.(x, y)
 
-    # TODO: why does this crash Julia?
-    # function slow_task!((x, digits, n), j, k)
-    #     start = 1 + (n * (j - 1)) ÷ num_threads()
-    #     stop =  (n* k) ÷ num_threads()
-    #     target = 0.0
-    #     for i ∈ start:stop
-    #         target += occursin(digits, string(i)) ? 0.0 : 1.0 / i
-    #     end
-    #     x[1,j] = target
-    # end
-    # function slow_cheap(n, digits)
-    #     x = zeros(8, num_threads())
-    #     batch(slow_task!, (num_threads(), num_threads()), x, digits, n)
-    #     sum(@view(x[1,1:end]))
-    # end
-    # slow_cheap(1000, "9")
+    function slow_task!((x, digits, n), j, k)
+        start = 1 + (n * (j - 1)) ÷ num_threads()
+        stop =  (n* k) ÷ num_threads()
+        target = 0.0
+        for i ∈ start:stop
+            target += occursin(digits, string(i)) ? 0.0 : 1.0 / i
+        end
+        x[1,j] = target
+    end
+    function slow_cheap(n, digits)
+        x = zeros(8, num_threads())
+        batch(slow_task!, (num_threads(), num_threads()), x, digits, n)
+        sum(@view(x[1,1:end]))
+    end
+    slow_cheap(1000, "9")
```

Comment:
Should probably use `@test` (vs a non-threaded version)

Reply:
Yep good point!

Reply:
```julia
  Expression: slow_cheap(1000, "9") ≈ slow_single_thread(1000, "9")
   Evaluated: 6.589720190283038 ≈ 6.590720190283038
```

Reply:
I wrote the code, so the but is probably my fault.

Reply:
That's strange, I'm getting:
 ```Julia
Julia> @test slow_cheap(1000, "9") ≈ slow_single_thread(1000,"9")
Test Passed
julia> slow_cheap(1000, "9")
6.5907201902830375

julia> slow_single_thread(1000,"9")
6.590720190283038
```

Reply:
Oh I've got a change in my local copy. I was gonna do it separately but can include here - the start value is wrong here
https://github.com/JuliaSIMD/CheapThreads.jl/blob/b02d71a5765ff568fe7e56e9ed5e69b638ce2c4c/src/batch.jl#L76
It needs a plus one, as per
https://github.com/JuliaSIMD/CheapThreads.jl/blob/b02d71a5765ff568fe7e56e9ed5e69b638ce2c4c/src/batch.jl#L8

Reply:
```julia
julia> slow_single_thread(cld(35*1000,36),"9")
6.589720190283038

julia> slow_cheap(1000,"9")
6.589720190283038

julia> Sys.CPU_THREADS
36
```

Reply:
Great!
Just pulled your last commit;
```julia
julia> slow_cheap(1000,"9")
6.590720190283038

julia> slow_single_thread(1000,"9")
6.590720190283038
```

Reply:
Awesome. I can add an extra test for the values of start and stop if you like?

Reply:
More tests the merrier.